riscv/interactive.cc

```diff
@@ -218,6 +220,46 @@ freg_t sim_t::get_freg(const std::vector<std::string>& args)
   return p->get_state()->FPR[r];
 }
 
+void sim_t::interactive_vreg(const std::string& cmd, const std::vector<std::string>& args)
+{
+  if (args.size() == 1) {
+    // Show all the regs!
+    processor_t *p = get_core(args[0]);
+    int vlen = (int)(p->VU.get_vlen()) >> 3;
+    int elen = (int)(p->VU.get_elen()) >> 3;
+    int num_elem = vlen/elen;
+    fprintf(stderr, "VLEN=%d bits; ELEN=%d bits\n", vlen << 3, elen << 3);
+
+    for (int r = 0; r < NVPR; ++r) {
+        fprintf(stderr, "%-4s: ", vr_name[r]);
+        for (int e = num_elem-1; e >= 0; --e){
+            long unsigned val;
+            switch(elen){
+                case 8:
+                    val = ((long unsigned*)p->VU.reg_file + r*num_elem)[e];
```

Comment:
This isn't 32-bit safe.  I know the int128_t stuff isn't supported by GCC on 32-bit targets, but at least that one will trigger a compiler error.

Reply:
We have notice this and discuss the same issue [here](https://github.com/riscv/riscv-isa-sim/pull/303#pullrequestreview-248257156)
I add an __int128_t type check in autocond and report error at building stage.

Reply:
Maybe `uint64_t` instead of `long unsigned` could provide better portability and readability?

Reply:
Agree with kito.  I will refine them with precise integer type.